central/image/service/service_impl.go

```diff
@@ -508,6 +508,26 @@ func (s *serviceImpl) EnrichLocalImageInternal(ctx context.Context, request *v1.
 	return internalScanRespFromImage(img), nil
 }
 
+// scannerV4MatchRequest will return true if the request is for matching vulnerabilities to
+// components produced by the Scanner V4 indexer, false otherwise.
+func scannerV4MatchRequest(request *v1.EnrichLocalImageInternalRequest) bool {
+	// If indexer version is NOT empty, then assume the components are from the
+	// Scanner V4 indexer.
+	return request.GetIndexerVersion() != ""
+}
+
+func (s *serviceImpl) enrichWithVulnerabilities(img *storage.Image, request *v1.EnrichLocalImageInternalRequest) error {
+	if !scannerV4MatchRequest(request) {
+		log.Debugf("Matching vulns for Clairify index report for image %q", img.GetName().GetFullName())
+		_, err := s.enricher.EnrichWithVulnerabilities(img, request.GetComponents(), request.GetNotes())
+		return err
+	}
+
+	log.Debugf("Matching vulns for Scanner V4 index report (ver: %s) for image %q", request.GetIndexerVersion(), img.GetName().GetFullName())
+	_, err := s.enricher.EnrichWithVulnerabilities(img, request.GetV4Contents(), nil)
```

Comment:
We should still have scan notes (ROX-21317)

Reply:
Yup, I can change this to pass the notes to the scannerv4 client - because the package/type of notes from sensor are `github.com/stackrox/scanner/generated/scanner/api/v1` I assumed we were not going to use them as is for scanner v4 and would need some conversion along the way - so I left it out.